### PR TITLE
Fix _mm256_setr_m128i related GNUC bug

### PR DESCRIPTION
**Pull request by lzhangnj**
Related to the MACRO definition of _mm256_setr_m128i under __GNUC__,
which was wrong (with lo and hi reversed) and already fixed in previous commit,
but forgot to clean the __GNUC__ work around in calling part.

I didn't create PR from hguermaz's fork, which has a blank line divergence in touched code base. The same fix has been in hguermaz::dev-03 in commit edfdf772.

### CODE REVIEW

File: Source/Lib/Common/ASM_AVX2/EbComputeSAD_Intrinsic_AVX2.c [hunk 1]
```diff
@@ -3960,7 +3958,6 @@ void get_eight_horizontal_search_point_results_32x32_64x64_pu_avx2_intrin(
     // ss3: Search Pos 3,7 for blocks 0,1,2,3
 
     ss4 = _mm256_cmpgt_epi32(ss0, ss1);
-    // not different printf("%d\n", _mm_extract_epi32(_mm256_extracti128_si256(ss4, 0), 0)); // DEBUG
     //ss4 = _mm256_or_si256(_mm256_cmpgt_epi32(ss0, ss1), _mm256_cmpeq_epi32(ss0, ss1));
```
Comment (mafonsoN) [inline Source/Lib/Common/ASM_AVX2/EbComputeSAD_Intrinsic_AVX2.c]: Perhaps you could remove the rest of the commented code?

**Comment by hassount** [comment]
@lzhangnj Could you please address the pending discussion items ?

File: Source/Lib/Common/ASM_AVX2/EbComputeSAD_Intrinsic_AVX2.c (hunk 1 above)
Comment (lzhangnj) [reply Source/Lib/Common/ASM_AVX2/EbComputeSAD_Intrinsic_AVX2.c]: Sorry I missed your messages.
This PR could change the encoding results when GNUC compilers are used.

**Comment by hassount** [comment]
@lzhangnj can you please try to rebase this to see if the travis build would pass ?

**Pull request merged**